### ISSUE TITLE
Change data definition format

### ISSUE THREAD

**Issue by rmarx** [quicwg/qlog#143]
We currently use a custom DDL inspired by TypeScript to specify events.
This should probably be changed to a format already defined by the IETF. This would also potentially remove the need to specify a custom JSON mapping.

Currently, YANG and especially CDDL seem to be the most attractive options for this purpose. 

Existing discussion was mainly in https://mailarchive.ietf.org/arch/msg/quic/H_inwB1oLh4fr1-i7JyUb-8U5Gg

**Comment by nibanks** [comment]
I have the following comments on format:

1. If it's not binary (or extremely efficient to generate and process even at extremely large scale) then I can't use it at run time. The best I can do is support a post processor to convert to qlog later. The tools will likely have problem with large files as well.
2. IMO, it **must** be simple to write a custom converter to the format. Please no super complicated protocols just around format. If I have to include an external library just to deal with the file format, then it's likely it won't be supported.

**Comment by rmarx** [comment]
To clarify, there are two types of formats we need to consider: 
1. The data definition format: how we define events and member data types in the drafts (that's what this issue is intended to be about)
2. The serialization format(s): what a qlog file actually looks like in string/binary form (I realized we no longer had an issue to discuss that, so I've made https://github.com/quiclog/internet-drafts/issues/144)

I think your comments mainly reflect on the 2nd point, though they are of course somewhat tied: for example, CDDL makes it easy to express JSON and CBOR, but potentially more difficult to define other (binary) serialization formats.
